src/vs/workbench/api/common/extHostWorkspace.ts

```diff
@@ -573,6 +576,37 @@ export class ExtHostWorkspace implements ExtHostWorkspaceShape, IExtHostWorkspac
 			this._onDidGrantWorkspaceTrust.fire();
 		}
 	}
+
+	// --- edit sessions ---
+
+	// called by ext host
+	registerEditSessionIdentityProvider(scheme: string, provider: vscode.EditSessionIdentityProvider) {
+		this._EditSessionIdentityProviders[scheme] = provider;
```

Comment:
Does the `scheme` need to be uri-transformed? E.g should `vscode-remote` become `file` and vice versa?

(we do this for language features that are registered per [scheme](https://github.com/microsoft/vscode/blob/52b6278e73d59fd5625e7c38e935007599c5d806/src/vs/workbench/api/common/extHostTypeConverters.ts#L164))

Reply:
Should you throw when registering for schema that's already taken? (I see you do that in the underlying service, which is good, but it is also nice to throw sync for when a provider is registered in the same extension host already)